### PR TITLE
docs: Provide operatorhub.io installation instructions

### LINKED ISSUE #779: Bring the SPO to operatorhub.io
#### What would you like to be added:
Adding the operator to [Operator Hub](https://operatorhub.io/) to automate the lifecycle of the operator when installed in OpenShift.

#### Why is this needed:

It would be a good addition to spread the usage of the operator while making its installation and upgrade easier.

#### User story covered

None so far.

### PR DESCRIPTION
**Pull request by jhrozek**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Now that SPO exists on operatorhub.io, make sure we tell people how to
use it.

#### Which issue(s) this PR fixes:
Fixes #779

#### Does this PR have test?
N/A

#### Special notes for your reviewer:
Tested on our vagrant fedora image.

#### Does this PR introduce a user-facing change?

```release-note
It is now possible to install SPO from packages provided on operatorhub.io. User-facing documentation
is provided in the installation-usage.md document.
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by jhrozek** [comment]
@saschagrunert I'm sorry to have invalidated the lgtm tag, but I also added instructions on using the upstream bundles. This is mainly useful for people wanting to test the latest code. See also one question inline

File: examples/olm/install-resources.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ metadata:
   namespace: olm
 spec:
   sourceType: grpc
-  image: quay.io/security-profiles-operator/security-profiles-operator-catalog:latest
+  image: gcr.io/k8s-staging-sp-operator/security-profiles-operator-catalog:latest
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply examples/olm/install-resources.yaml:15]: Yes, definitely :+1:

**Comment by jhrozek** [comment]
I amended the release doc in this PR since the tests needed a re-run anyway :-)

**Review by saschagrunert — APPROVED** [review]

**Comment by jhrozek** [comment]
Rebased on master in hopes of getting a clean CI run

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
Merging to save the environment.

**Pull request merged**